com.unity.multiplayer.mlapi/Tests/Runtime/Transport/SIPTransport.cs

```diff
@@ -220,38 +220,52 @@ public override SocketTasks StartServer()
 
     public override void Send(ulong clientId, ArraySegment<byte> data, NetworkChannel channel)
     {
-        // Create copy since MLAPI wants the byte array back straight after the method call.
-        // Hard on GC.
-        byte[] copy = new byte[data.Count];
-        Buffer.BlockCopy(data.Array, data.Offset, copy, 0, data.Count);
-
-        m_Clients[clientId].IncomingBuffer.Enqueue(new Event
+        if (m_LocalConnection != null)
```

Comment:
If a player is disconnected m_LocalConnection becomes null, but can still be called since the NetworkManager is not notified of the client disconnecting.  This resolves this issue.

Reply:
this "skip if null" logic might simply make us blind to some issues happening in the future.
please log a warning on the else branch at least.

Reply:
@TwoTenPvP Please review the above changes.
SIPTransport.DisconnectLocalClient sets m_LocalConnection to null but never notifies the relative NetworkManager that the client has disconnected which can cause other systems (i.e. SnapShotSystem) to cause a barrage of exceptions when trying to send snapshots from a disconnected client.

Reply:
It is meant to notify all peers that it disconnected.

Reply:
I don't know why this would happen or what's going on behind the scenes but I still want to see a warning log printed so that a dev wouldn't be blind and execution wouldn't fail silently. do you guys think my request is still reasonable?

Reply:
Let me DM you on this... there are other things to consider here.